src/main/java/com/cisco/trex/stateful/TRexAstfClient.java

```diff
@@ -407,48 +403,52 @@ public String getVersion() {
      * @return
      */
     public Map<String, AstfStatistics> getTemplateGroupStatistics(List<String> tgNames) {
-        return getTemplateGroupStatistics(null, tgNames);
+        return getTemplateGroupStatistics("", tgNames);
     }
 
     /**
      * get template group statistics
      *
      * @param profileId
      * @param tgNames
-     * @return
+     * @return Map key:tgName, value:AstfStatistics
      */
     public Map<String, AstfStatistics> getTemplateGroupStatistics(String profileId, List<String> tgNames) {
         Map<String, AstfStatistics> stats = new HashMap<>();
         Map<String, Object> payload = createPayload(profileId);
         payload.put("epoch", 1);
 
+        //remove duplicated tgNames in input list
+        tgNames = new ArrayList<>(new HashSet<>(tgNames));
         Map<String, Integer> name2Id = translateNames2Ids(profileId, tgNames);
         List<Integer> tgIds = new ArrayList<>(tgNames.size());
         tgNames.forEach(name -> tgIds.add(name2Id.get(name)));
         payload.put("tg_ids", tgIds);
 
         String json = callMethod("get_tg_id_stats", payload);
         JsonElement response = new JsonParser().parse(json);
-
         JsonObject result = response.getAsJsonArray().get(0).getAsJsonObject().get("result").getAsJsonObject();
-        Map<Integer, String> id2Name = translateIds2Names(name2Id);
-        for (Integer id : tgIds){
-            stats.put(id2Name.get(id), GSON.fromJson(result.get(id.toString()), AstfStatistics.class));
-        }
+        name2Id.forEach((key, value) ->{
+            try {
+                AstfStatistics astfStatistics = new ObjectMapper().readValue(result.get(value.toString()).toString(), AstfStatistics.class);
+                astfStatistics.setCounterNames(getAstfStatsMetaData());
+                stats.put(key, astfStatistics);
+            } catch (IOException e) {
+//                e.printStackTrace();
```

Comment:
catch (IOException e) {
            LOGGER.error("Error occurred during processing output  of get_tg_id_stats method, e);
        }